### DISCUSSION TITLE
Using `getLocaleFromRoute` from within `defineNuxtRouteMiddleware`

### DISCUSSION THREAD

**Discussion by rudolfbyker** [nuxt-modules/i18n#3074] [Q&A]
Is it possible to get a `getLocaleFromRoute` function within my custom Nuxt route middleware?

This import will not work:

```ts
import { createLocaleFromRouteGetter } from "../node_modules/@nuxtjs/i18n/dist/runtime/routing/extends/router.js";
```

```
Package subpath './dist/runtime/routing/extends/router' is not defined by "exports" in .../node_modules/@nuxtjs/i18n/package.json
```

I don't just want the "current locale". Rather, I want a function that will reliably give me the locale, given an arbitrary route.

My current workaround is to re-implement `getLocaleFromRoute` myself.

**Comment by frederikheld** [comment]
+1, I need that myself. Currently I'm using a regex to extract the locale key from the route, but this will only work as long as the strategy stays the same. I'd like to have something that works reliably no matter what is configured.
